openshift-kube-controller-manager/servicecacertpublisher/publisher.go

```diff
@@ -155,18 +158,52 @@ func (c *Publisher) processNextWorkItem() bool {
 	return true
 }
 
+var (
+	// default secure
+	// This annotation prompts the service ca operator to inject
+	// the service ca bundle into the configmap.
+	injectionAnnotation = map[string]string{
+		"service.beta.openshift.io/inject-cabundle": "true",
+	}
+	setAnnotationOnce = sync.Once{}
+)
+
+func getInjectionAnnotation() map[string]string {
+	setAnnotationOnce.Do(func() {
+		// this envvar can be used to get the kube-controller-manager to inject a vulnerable legacy service ca
+		// the kube-controller-manager carries no existing patches to launch, so we aren't going add new
+		// perma-flags.
+		// it would be nicer to find a way to pass this more obviously.  This is a deep side-effect.
+		// though ideally, we see this age out over time.
+		useVulnerable := os.Getenv("OPENSHIFT_USE_VULNERABLE_LEGACY_SERVICE_CA_CRT")
+		if len(useVulnerable) == 0 {
+			return
+		}
+		useVulnerableBool, err := strconv.ParseBool(useVulnerable)
+		if err != nil {
+			// caller went crazy, don't use this unless you're careful
```

Comment:
nit: I'm not able to discern your intent from this comment, maybe restate?

(No action required) Given that `ParseBool` will return an error if the value is not something that can be converted to a boolean, maybe emit a more explicit failure message (i.e. `Value supplied for OPENSHIFT_USE_VULNERABLE_LEGACY_SERVICE_CA_CRT cannot be parsed as a boolean: %v`) so that the solution to the error is obvious from the log entry alone?

Reply:
> (No action required) Given that `ParseBool` will return an error if the value is not something that can be converted to a boolean, maybe emit a more explicit failure message (i.e. `Value supplied for OPENSHIFT_USE_VULNERABLE_LEGACY_SERVICE_CA_CRT cannot be parsed as a boolean: %v`) so that the solution to the error is obvious from the log entry alone?

You are correct. I'm hoping to see more refinement in this area in either 4.9 or 4.10.  Perhaps by removing the KCM patch entirely.  This is ugly, but it will leave a stack line and it means the operator failed.  I'd like to leave this as-is without having error in the return path for backporting.